pkg/function/restore_csi_snapshot_test.go

```diff
@@ -0,0 +1,192 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package function
+
+import (
+	"context"
+
+	"github.com/kanisterio/kanister/pkg/kube/snapshot"
+
+	. "gopkg.in/check.v1"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	dynfake "k8s.io/client-go/dynamic/fake"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+const (
+	// RestoreCSISnapshotTestNamespace is the namespace where testing is done
+	RestoreCSISnapshotTestNamespace = "test-restore-csi-snapshot"
+	// RestoreCSISnapshotOriginalPVCName is the name of the PVC that will be captured
+	RestoreCSISnapshotOriginalPVCName = "test-pvc"
+	// RestoreCSISnapshotPVCName is the name of the new PVC that will be restored
+	RestoreCSISnapshotNewPVCName = "test-pvc-restored"
+	// RestoreCSISnapshotSnapshotName is the name of the snapshot
+	RestoreCSISnapshotSnapshotName = "test-snapshot"
+	// RestoreCSISnapshotSnapshotClass is the fake snapshot class
+	RestoreCSISnapshotSnapshotClass = "test-snapshot-class"
+	// RestoreCSISnapshotStorageClass is the fake storage class
+	RestoreCSISnapshotStorageClass = "test-storage-class"
```

Comment:
Do we need to export these?

Reply:
No we don't. These are only used in the following test cases.